source/release-notes.txt

```diff
@@ -20,6 +20,21 @@ Release Notes for |bi|
 .. expect and we don't break the links on the downloads page.
 .. (DOCS-9670 for context)
 
+.. _bi-2-14-10:
+
+|bi| 2.14.10
+------------
+
+*Released August 7, 2023*
+
+Bug Fixes
```

Comment:
Similar to the 2.14.9 notes, let's add an "improvement" that we now build with go 1.20.6 to fix CVE-2023-29409